message/infrastructure/sql/schema_adapter.go

```diff
@@ -0,0 +1,122 @@
+package sql
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+// SchemaAdapter produces the SQL queries and arguments appropriately for a specific schema and dialect
+// It also transforms sql.Rows into Watermill messages.
+type SchemaAdapter interface {
+	// InsertQuery returns the SQL query that will insert the Watermill message into the SQL storage.
+	InsertQuery(topic string, msgs message.Messages) (string, []interface{}, error)
```

Comment:
The comments under `OffsetsAdapter` apply here as well

Reply:
done